pygls/workspace.py

```diff
@@ -97,16 +91,63 @@ def _apply_incremental_change(self, change: TextDocumentContentChangeEvent) -> N
         self._source = new.getvalue()
 
     def _apply_full_change(self, change: TextDocumentContentChangeEvent) -> None:
-        """Apply a full text change to the document."""
+        """Apply a FULL text change to the document."""
         self._source = change.text
 
+    def _apply_none_change(self, change: TextDocumentContentChangeEvent) -> None:
+        """Apply a NONE text change to the document
+
+        Currently does nothing, provided for consistency.
+        """
+
     def apply_change(self, change: TextDocumentContentChangeEvent) -> None:
-        """Apply change to a document, depending on synchronization kind."""
-        try:
-            self._sync_func(change)
-        except TypeError:
-            # Don't sync document if sync kind is TextDocumentSyncKind.NONE
-            pass
+        """Apply a text change to a document, considering TextDocumentSyncKind
+
+        Performs either INCREMENTAL, FULL, or NONE synchronization based on
+        both the Client request and server capabilities.
+
+        Raises ValueError if the client provides an ambiguous or unsupported
```

Comment:
`ValueError` is not raised in this method, please remove this part of _docstring_.